### ISSUE TITLE
member-ordering is not enforced when private variable goes after optional public variable

### ISSUE THREAD

**Issue by pronebird** [palantir/tslint#4506]
### Bug Report

- __TSLint version__:  5.1.2
- __TypeScript version__: 3.x
- __Running TSLint via__: CLI

#### TypeScript code being linted

TSLint does not consider the `d` member as being misplaced. (I haven't tested this exact piece though)

```ts
class A {
  set b(newValue: boolean) {
     // [redacted]
  }

  get b(): boolean {
    return false;
  }

  public c?: (string) => void;

  private d = {
    whatever: true,
  };

  public e() {
    // redacted
  }
}
```

Moving `d` above `c` does in fact trigger the member ordering rule:

>member-ordering  Declaration of public instance field not allowed after declaration of private instance field. Instead, this should come at the beginning of the class/interface.

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:latest",
    "tslint-react",
    "tslint-config-prettier"
  ],
  "rules": {
    "max-classes-per-file": false,
    "object-literal-sort-keys": false,
    "variable-name": [
      true,
      "ban-keywords",
      "check-format",
      "allow-pascal-case",
      "allow-leading-underscore"
    ]
  }
}
```

#### Actual behavior

Member order is not enforced even though the private variable is trapped between public variable and a method.

#### Expected behavior

That the member order would be enforced

**Comment by pronebird** [comment]
I've realised that tslint treats instance variables and methods differently so that's the correct behaviour.
